### PR TITLE
CURA-6632 Add intent into stats

### PR DESCRIPTION
**Pull request by LipuFei**

### CODE REVIEW

File: cura/Settings/GlobalStack.py [hunk 1]
```diff
@@ -131,6 +131,14 @@ def getConfigurationTypeFromSerialized(cls, serialized: str) -> Optional[str]:
             return "machine_stack"
         return configuration_type
 
+    def getIntentCategory(self) -> str:
+        intent_category = "default"
+        for extruder in self.extruderList:
+            category = extruder.intent.getMetaDataEntry("intent_category", "default")
+            if category != "default" and category != intent_category:
+                intent_category = category
+        return intent_category
```
Comment (Ghostkeeper) [inline cura/Settings/GlobalStack.py]: This code duplicates `MachineManager.activeIntentCategory`. We need to remove either one.

Comment (LipuFei) [reply cura/Settings/GlobalStack.py]: `MachineManager.activeIntentCategory` calls this function to get the intent cateogry. It also checks if the active global stack is present and always returns a string.

File: cura/Settings/MachineManager.py [hunk 2]
```diff
@@ -622,12 +622,7 @@ def activeIntentCategory(self) -> str:
 
         if not global_container_stack:
             return ""
-        intent_category = "default"
```
Comment (nallath) [inline cura/Settings/MachineManager.py]: This change is breaking one of the tests. So either this change is wrong, or the test is wrong.

Comment (LipuFei) [reply cura/Settings/MachineManager.py]: Fixed by rebasing on the latest `feature_intent` branch.

Comment (ninovanhooff) [reply cura/Settings/MachineManager.py]: Confirmed: no more failing tests

**Pull request merged**